### DISCUSSION TITLE
Blocking bots from server

### DISCUSSION THREAD

**Discussion by SpeedyCharly** [AzuraCast/AzuraCast#7360] [Ideas]
As you are well aware, in Azuracast dashboard, under REPORTS, we have access to a page listing all connected listeners.  It's usefulness is limited to seeing the composition and location of our listener base. 
Our station is currently assailed by legions of bots, eating our bandwith and probably illegally scraping our content as I have reported in an earlier discussion (See; https://github.com/AzuraCast/AzuraCast/discussions/7342).  The only effective means of blocking such bots is to manually block their ip via SSL (iptables... Drop) wich is extremely time consuming because such bots use VPN and tend to come back on a different ip.

In a bid to make the process a bit easier, would it be possible to have a switch on this page where we could block such abusive users with just 1 click? 

Thanks;
Speedy

**Comment by gunsar** [comment]
try
**Edit Station Profile :** 
**Banned User Agents :**
fill in the bot you want to ban, for example:
`*yandex*`
`*bingbot*`
`*Pinterestbot*`
**save and restart station**

**Reply by SpeedyCharly** [reply]
Will try that thanks!

**Reply by SpeedyCharly** [reply]
Doesn't seem to work. Here is example of the bots I get;
---------------------------
147.182.185.159	
Bot/Crawler 
zgrab (Security Checker)
Mozilla/5.0 zgrab/0.x

69.171.249.15	
Facebook Crawler (Social Media Agent)
facebookexternalhit/1.1 (+http://www.facebook.com/externalhit_uatext.php)

148.113.182.166		
Bot/Crawler
Generic Bot (Generic Crawler)
SSL Proxy for client IP: 18.192.14.156 / Country: DE
-------------------------------
Tried inputting as Facebook Crawler, facebookexternalhit, etc.  but they keep coming back

**Comment by SpeedyCharly** [comment]
Coming back to this thread once again as nothing seems to work to try and block bot from accessing the stream;
Blocked about 10 ip ranges via SSL yesterday [iptables -A INPUT -s xx.yyy.zzz.0/24 -j DROP] and still have 10+ bots today using the same ip ranges...
And Gunsar's suggestion doesn't work either... unless I'm not using proper syntax...

Any help appreciated - tired of having these bots steal our bandwidth all the time !

**Reply by BusterNeece** [reply]
@SpeedyCharly You may want to try our new bot protection in the latest Rolling Release. If you set `NGINX_BLOCK_BOTS=1` in `azuracast.env` (or answer "Yes" when prompted about it during a CLI update), we'll automatically download and update the nginx ultimate bot blocker for your installation, which stops a fair number of scrapers and bad actors from crawling your site.

**Reply by SpeedyCharly** [reply]
Was on rolling release for a long while at first only to realize that I'm not savvy enough on Linux to deal with the boo-boos that inevitably occur.  Which is why I've been on stable for some time now.

Any hopes of seeing this in a stable release soon?

**Reply by BusterNeece** [reply]
In due time. Too many changes to the scheduler to promise a stable release in the immediate future.
